src/effects/VST3/VST3Effect.cpp

```diff
@@ -331,82 +331,68 @@ bool VST3Effect::SupportsRealtime() const
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+
+   return AnyOf
+   (
+      [&](const ParameterInfo& parameterInfo)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
-            return true;
+         return (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate);
       }
-   }
-
-   return false;
+   );
 }
 
 bool VST3Effect::SaveSettings(
-   const EffectSettings &, CommandParameters & parms) const
+   const EffectSettings& settings, CommandParameters & parms) const
 {
-   if(mEditController == nullptr)
+   if (mEditController == nullptr)
       return false;
 
+   const VST3EffectSettings& ownSettings = GetSettings(settings);
+
    using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+
+   return ForEachParameter
+   (
+      [&](const ParameterInfo& parameterInfo)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
          {
             parms.Write(
                VST3Utils::MakeAutomationParameterKey(parameterInfo),
-               mEditController->getParamNormalized(parameterInfo.id)
+               ownSettings.mValues.at(parameterInfo.id)
             );
          }
+         return true;
       }
-   }
-
-   return true;
+   );
 }
 
 bool VST3Effect::LoadSettings(
    const CommandParameters & parms, EffectSettings &settings) const
 {
    using namespace Steinberg;
 
-   if(mComponentHandler == nullptr)
-      return false;
-   
-   long index { };
-   wxString key;
-   if(parms.GetFirstEntry(key, index))
-   {
-      do
+   auto& ownSettings = GetSettings(settings);
+   ownSettings.mValues.clear();
+
+   ForEachParameter
+   (
+      [&](const ParameterInfo& parameterInfo)
       {
-         Steinberg::Vst::ParamID id;
-         Vst::ParamValue value;
-         if(VST3Utils::ParseAutomationParameterKey(key, id) && parms.Read(key, &value))
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
          {
-            if(mComponentHandler->beginEdit(id) == kResultOk)
+            const wxString key = VST3Utils::MakeAutomationParameterKey(parameterInfo);
+            Vst::ParamValue value;
+
+            if (parms.Read(VST3Utils::MakeAutomationParameterKey(parameterInfo), &value))
             {
-               auto cleanup = finally([&]{
-                  mComponentHandler->endEdit(id);
-               });
-               mComponentHandler->performEdit(id, value);
+               ownSettings.mValues[parameterInfo.id] = value;;
             }
-            mEditController->setParamNormalized(id, value);
          }
-      } while(parms.GetNextEntry(key, index));
-   }
-
-   if(mPlainUI != nullptr)
-      mPlainUI->ReloadParameters();
```

Comment:
Strictly speaking, the deletion of these two lines ought to wait until the commit "Define TransferDataToWindow" so that each step if the transformation is correct, introducing no temporary known bugs.